homeassistant/components/airvisual/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Define a config flow manager for AirVisual."""
+from collections import OrderedDict
+
+from pyairvisual import Client
+from pyairvisual.errors import InvalidKeyError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_SHOW_ON_MAP,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+
+from .const import CONF_GEOGRAPHIES, DOMAIN
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured AirVisual instances."""
+    return set(
+        entry.data[CONF_API_KEY] for entry in hass.config_entries.async_entries(DOMAIN)
+    )
+
+
+class AirVisualFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a AirVisual config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @property
+    def cloud_api_schema(self):
+        """Return the data schema for the cloud API."""
+        schema = OrderedDict()
+        schema[vol.Required(CONF_API_KEY)] = str
+        schema[
+            vol.Required(CONF_LATITUDE, default=self.hass.config.latitude)
+        ] = cv.latitude
+        schema[
+            vol.Required(CONF_LONGITUDE, default=self.hass.config.longitude)
+        ] = cv.longitude
+        schema[vol.Optional(CONF_SHOW_ON_MAP, default=True)] = bool
```

Comment:
Keep this out of the config entry and keep a sane default value until options flow is available